core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3296,6 +3298,13 @@ class KafkaApis(val requestChannel: RequestChannel,
     handle(forwardedRequest)
   }
 
+  def handleDecommissionBrokerRequest(request: RequestChannel.Request): Unit = {
+    val decommissionBrokerRequest = request.body[DecommissionBrokerRequest]
+    requestHelper.sendResponseMaybeThrottle(request, requestThrottleMs =>
+      decommissionBrokerRequest.getErrorResponse(requestThrottleMs,
+        Errors.UNSUPPORTED_VERSION.exception))
```

Comment:
If we advertise this API in the ApiVersions response, then it would be surprising to get an UNSUPPORTED_VERSION error. I think we might need augment the ApiVersion handler here so that we are able to exclude KIP-500 APIs when it is not enabled.

Reply:
That makes sense.